aalaesar/install_nextcloud
# Multiple changes and gitlab-ci testing

Link to testing pipelines:
https://gitlab.ninux.org/hispanico/ansible-nextcloud/pipelines?scope=branches

Changes:
- Replaced all 'include' for task with 'include_tasks' to fix DEPRECATION WARNING in ansible 2.4
- Configured NC config.php manipolation tasks with occ command
- Added redis server installation
- Added copy of file mimetypemapping.json to set custom mimetypemapping (es. ownpad app file extension)
- Disebled apache mod_ssl when tls is false
- Added nextcloud settings:
   - trusted proxies list
   - redis
   - backgroud jobs with cron
   - custom settings

- Added mysql settings for utf8mb4:
   - innodb_large_prefix=on
   - innodb_file_format=barracuda
   - innodb_file_per_table=true

- Created .gitlab-ci.yml for testing
   - Ubuntu Xenial
   - Ubuntu Trusty
   - Debian Stretch
   - Debian Jessie

- Created testing playbooks
   - nextcloud10_mariadb.yml
   - nextcloud10_mysql.yml
   - nextcloud10_nginx.yml
   - nextcloud10_psql.yml
   - nextcloud_mariadb.yml
   - nextcloud_mysql.yml
   - nextcloud_nginx.yml
   - nextcloud_psql.yml

- Update README.md

- Set just ansible syntax-check in travis
